src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2556,6 +2556,11 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
         } else {
             $mailer->replaceArgument(1, $messageBus ? new Reference($messageBus) : new Reference('messenger.default_bus', ContainerInterface::NULL_ON_INVALID_REFERENCE));
         }
+        $authenticators = [];
+        foreach ($config['smtp']['authenticators'] ?? [] as $authenticator) {
+            $authenticators[] = new Reference($authenticator);
+        }
+        $container->getDefinition('mailer.transport_factory.smtp')->setArgument(3, $authenticators);
```

Comment:
according to `mailer_transport.php` this service is the EsmtpTransportFactory, but the CI tells that we somehow end up setting an argument to "Symfony\Component\Mailer\Mailer" - any idea what i am messing up here?

Reply:
is that a bug of the DI container, or should i be doing something differently?

Reply:
I would not call it a bug. The hard thing is that inline definitions don't have an id. So that's hard to know how to mention them in error messages.